src/V3SplitVar.cpp

```diff
@@ -0,0 +1,1258 @@
+            virtual void visit(AstNode* nodep) VL_OVERRIDE { iterateChildren(nodep); }
+            virtual void visit(AstVar* nodep) VL_OVERRIDE { m_parent.m_vars.erase(nodep); }
+            virtual void visit(AstVarRef* nodep) VL_OVERRIDE { m_parent.m_refs.erase(nodep); }
+            virtual void visit(AstSel* nodep) VL_OVERRIDE {
+                m_parent.m_sels.erase(nodep);
+                iterateChildren(nodep);
+            }
+            explicit Visitor(RefsInModule& p)
+                : m_parent(p) {}
+        } v(*this);
+        v.iterate(nodep);
+    }
+    void visit(AstNVisitor* visitor) {
+        for (std::set<AstVar*>::iterator it = m_vars.begin(), it_end = m_vars.end(); it != it_end;
+             ++it) {
+            visitor->iterate(*it);
+        }
+        for (std::set<AstSel*>::iterator it = m_sels.begin(), it_end = m_sels.end(); it != it_end;
+             ++it) {
+            // If m_refs includes VarRef from ArraySel, remove it
+            // because the VarRef would not be visited in SplitPackedVarVisitor::visit(AstSel*).
+            if (AstVarRef* refp = VN_CAST((*it)->fromp(), VarRef)) {
+                m_refs.erase(refp);
+            } else if (AstVarRef* refp = VN_CAST((*it)->lsbp(), VarRef)) {
+                m_refs.erase(refp);
+            } else if (AstVarRef* refp = VN_CAST((*it)->widthp(), VarRef)) {
+                m_refs.erase(refp);
+            }
+            UASSERT_OBJ(reinterpret_cast<uintptr_t>((*it)->op1p()) != 1, *it, "stale");
+            visitor->iterate(*it);
+        }
+        for (std::set<AstVarRef*>::iterator it = m_refs.begin(), it_end = m_refs.end();
+             it != it_end; ++it) {
+            UASSERT_OBJ(reinterpret_cast<uintptr_t>((*it)->op1p()) != 1, *it, "stale");
+            visitor->iterate(*it);
+        }
+    }
+};
+
+typedef std::set<AstVar*, AstNodeComparator> VarSet;
+typedef std::map<AstNodeModule*, RefsInModule, AstNodeComparator> RefsMap;
+
+class SplitUnpackedVarVisitor : public AstNVisitor {
+    VarSet m_foundTargetVar;
+    UnpackRefMap m_refs;
+    AstNodeModule* m_modp;
+    // AstNodeStmt, AstCell, AstNodeFTaskRef, or AstAlways(Public) for sensitivity
+    AstNode* m_context;
+    AstNodeFTask* m_inFTask;
+    size_t m_numSplit;
+    // List for SplitPackedVarVisitor
+    RefsMap m_refsForPackedSplit;
+
+    static AstVarRef* isTargetVref(AstNode* nodep) {
+        if (AstVarRef* refp = VN_CAST(nodep, VarRef)) {
+            if (refp->varp()->attrSplitVar()) return refp;
+        }
+        return NULL;
+    }
+    static int outerMostSizeOfUnpackedArray(AstVar* nodep) {
+        AstUnpackArrayDType* dtypep = VN_CAST(nodep->dtypep()->skipRefp(), UnpackArrayDType);
+        UASSERT_OBJ(dtypep, nodep, "Must be unapcked array");
+        return dtypep->msb() - dtypep->lsb() + 1;
+    }
+
+    void setContextAndIterateChildren(AstNode* nodep) {
+        const ContextKeeper keeper(&m_context, nodep);
+        iterateChildren(nodep);
+    }
```

Comment:
I can live with this, but would probably favor the more verbose but obvious

        AstNode* origContextp = m_context;
        {
           m_contextp = nodep;
            iterateChildren(nodep);
        }
        m_contextp = origContextp;
    }

(Which is why I never made a template to save/restore such things.)